superwall/src/main/java/com/superwall/sdk/paywall/vc/web_view/messaging/PaywallMessageHandler.kt

```diff
@@ -109,14 +122,24 @@ class PaywallMessageHandler(
             is PaywallMessage.Restore -> restorePurchases()
             is PaywallMessage.Purchase -> purchaseProduct(withId = message.productId)
             is PaywallMessage.PaywallOpen -> {
-                CoroutineScope(Dispatchers.IO).launch {
-                    pass(eventName = "paywall_open", paywall = paywall)
+                ioScope.launch {
+                    pass(eventName = SuperwallEvents.PaywallOpen.rawName, paywall = paywall)
+                }
+            }
+            is PaywallMessage.PaywallClose -> {
+                ioScope.launch {
+                    val eventName = SuperwallEvents.PaywallClose.rawName
+                    pass(eventName = eventName, paywall = paywall)
```

Comment:
This needs to be like iOS – if we pass messages to the paywall before the paywalljs has loaded then it'll result in an error. So on iOS we queue up the message and post it after the webview has loaded. Have a look at the equiv iOS class

Reply:
Added a `Queue` for that and for `PaywallOpen` so we have parity with iOS 👍